vmaas/reposcan/redhatoval/definitions_file.py

```diff
@@ -105,17 +111,32 @@ def _parse_definitions(self):
                         if affected_cpe_list is not None:
                             for child in affected_cpe_list.findall("default:cpe", NS):
                                 cpes.append(text_strip(child))
+                        affected = advisory.find("default:affected", NS)
+                        if affected is not None:
+                            for resolution in affected.findall("default:resolution", NS):
+                                if resolution.get("state", "") in UNSUPPORTED_RESOLUTION_STATES:
+                                    for component in resolution.findall("default:component", NS):
+                                        ignored_components.add(text_strip(component))
                 criteria = self._parse_criteria(definition.find("default:criteria", NS))  # <criteria> 0..1
 
                 # Parse tests
                 tests = []
                 crit = criteria
+                current_module = ""
                 criteria_stack = []
                 while crit is not None:
-                    tests.extend(crit["criterions"])
-                    criteria_stack.extend(crit["criteria"])
+                    for criterion in crit["criterions"]:
+                        parts = criterion["comment"].split()
+                        if parts and parts[0] == "Module" and parts[-1] == "enabled" and crit["operator"] == "AND":
+                            current_module = "%s/" % parts[1]
+                        if parts and "%s%s" % (current_module, parts[0]) in ignored_components:
```

Comment:
I think this is not correct because there is `Module` in `parts[0]` so from `"%s%s" % (current_module, parts[0])` we get `php:7.3/Module` instead of component like `php:7.3/php`

Reply:
Should be fixed now by the "else if"